### PR TITLE
RIDER-55107: Fix function annotation on let keyword

### PR DESCRIPTION
**Pull request by mcon**
Execution wasn't consistent with availability - failing test added, fix implemented.

Once this is in I'll update Once this is in, I'll update https://github.com/JetBrains/fsharp-support/pull/202 to target `net211` and fix up any conflicts with this change :+1:.

### CODE REVIEW

File: ReSharper.FSharp/src/FSharp.Psi.Features/src/Intentions/FunctionAnnotationAction.fs [hunk 1]
```diff
@@ -99,7 +99,8 @@ type FunctionAnnotationAction(dataProvider: FSharpContextActionDataProvider) =
         isAtLetExprKeywordOrNamedPat dataProvider letBindings && not (isAnnotated bindings.[0])
 
     override x.ExecutePsiTransaction _ =
-        let binding = dataProvider.GetSelectedElement<IBinding>()
+        let letBindings = dataProvider.GetSelectedElement<ILetBindings>()
+        let binding = letBindings.Bindings |> Seq.exactlyOne
```
Comment (auduchinok) [inline ReSharper.FSharp/src/FSharp.Psi.Features/src/Intentions/FunctionAnnotationAction.fs:103]: Wouldn't it fail on the following?
```fsharp
let rec f _ = ()
and g _ = ()
```

Comment (auduchinok) [reply ReSharper.FSharp/src/FSharp.Psi.Features/src/Intentions/FunctionAnnotationAction.fs:103]: Ah, it's cheched before, great.

Comment (mcon) [reply ReSharper.FSharp/src/FSharp.Psi.Features/src/Intentions/FunctionAnnotationAction.fs:103]: Yeah, just confirmed as much myself, feels like this is sufficient for what's currently available.

**Review by auduchinok — APPROVED** [review]
Thanks!

**Pull request merged**